uniffi_bindgen/src/bindings/swift/templates/ObjectTemplate.swift

```diff
@@ -42,14 +55,45 @@ public class {{ obj.name()|class_name_swift }}: {{ obj.name() }}Protocol {
 
     {%- when Some with (return_type) -%}
     public func {{ meth.name()|fn_name_swift }}({% call swift::arg_list_decl(meth) %}) {% call swift::throws(meth) %} -> {{ return_type|type_swift }} {
-        let _retval = {% call swift::to_ffi_call_with_prefix("self.handle", meth) %}
+        let _retval = {% call swift::to_ffi_call_with_prefix("self.pointer", meth) %}
         return {% call swift::try(meth) %} {{ "_retval"|lift_swift(return_type) }}
     }
 
     {%- when None -%}
     public func {{ meth.name()|fn_name_swift }}({% call swift::arg_list_decl(meth) %}) {% call swift::throws(meth) %} {
-        {% call swift::to_ffi_call_with_prefix("self.handle", meth) %}
+        {% call swift::to_ffi_call_with_prefix("self.pointer", meth) %}
     }
     {%- endmatch %}
     {% endfor %}
-}
\ No newline at end of file
+}
+
+fileprivate extension {{ obj.name()|class_name_swift }} {
+    fileprivate typealias FfiType = UnsafeMutableRawPointer
+
+    fileprivate static func read(from buf: Reader) throws -> Self {
+        let v: UInt64 = try buf.readInt()
+        // The Rust code won't compile if a pointer won't fit in a UInt64.
+        // We have to go via `UInt` because that's the thing that's the size of a pointer.
+        let ptr = UnsafeMutableRawPointer(bitPattern: UInt(truncatingIfNeeded: v))
+        if (ptr == nil) {
+            throw UniffiInternalError.unexpectedNullPointer
+        }
+        return try self.lift(ptr!)
+    }
+
+    fileprivate func write(into buf: Writer) {
+        // This fiddling is because `Int` is the thing that's the same size as a pointer.
+        // The Rust code won't compile if a pointer won't fit in a `UInt64`.
+        buf.writeInt(UInt64(bitPattern: Int64(Int(bitPattern: self.lower()))))
+    }
+
+    fileprivate static func lift(_ pointer: UnsafeMutableRawPointer) throws -> Self {
+        return Self(unsafeFromRawPointer: pointer)
+    }
+
+    fileprivate func lower() -> UnsafeMutableRawPointer {
+        return self.pointer
+    }
+}
+
+extension {{ obj.name()|class_name_swift }} : ViaFfi, Serializable {}
```

Comment:
I wasn't able to make it so, for reasons that I don't really understand to do with Swift's object model. Perhaps I'll take a fresh look at it now that I've been away from the code for a little while. At the very least, I can leave a comment here about *why* it can't be private.